crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -542,17 +578,66 @@ pub struct EntityLocation {
     pub index: usize,
 }
 
+// Constant for the initial generation value, removes the need to unwrap in opt-0 code but doesn't require unsafe.
+const GENERATION_ONE: NonZeroU32 = if let Some(gen) = NonZeroU32::new(1) {
+    gen
+} else {
+    panic!("Failed to unwrap GENERATION_ONE constant")
```

Comment:
```suggestion
    unreachable!("1≠0")
```

Reply:
Unfortunately `unreachable!` doesn't appear to be const even though `panic!` is 0.o

There's `std::hint::unchecked_unreachable()` but that requires `unsafe`

Reply:
I think a raw unreachable would be fine, otherwise some panic message like `unreachable: 1≠0` would also make sense.